monitoring/os_linux.go

```diff
@@ -0,0 +1,353 @@
+// https://www.freedesktop.org/software/systemd/man/os-release.html
+type OSChecker struct {
+	// Releases lists all supported releases
+	Releases   []OSRelease
+	getRelease osReleaseGetter
+}
+
+// Name returns name of the checker
+func (c *OSChecker) Name() string {
+	return osCheckerID
+}
+
+// Check checks current OS and release is within supported list
+func (c *OSChecker) Check(ctx context.Context, reporter health.Reporter) {
+	info, err := c.getRelease()
+	if err != nil {
+		reporter.Add(NewProbeFromErr(osCheckerID,
+			"failed to query OS version",
+			trace.Wrap(err)))
+		return
+	}
+
+	for _, release := range c.Releases {
+		if versionsMatch(release, *info) {
+			reporter.Add(&pb.Probe{
+				Checker: osCheckerID,
+				Status:  pb.Probe_Running,
+			})
+			return
+		}
+	}
+
+	reporter.Add(&pb.Probe{
+		Checker: osCheckerID,
+		Detail:  fmt.Sprintf("%s %s is not supported", info.ID, info.VersionID),
+		Status:  pb.Probe_Failed,
+	})
+}
+
+// OSRelease is used to represent a certain OS release
+// based on https://www.freedesktop.org/software/systemd/man/os-release.html
+type OSRelease struct {
+	// ID identifies the distributor: ubuntu, redhat/centos, etc.
+	ID string
+	// VersionID is the release version i.e. 16.04 for Ubuntu
+	VersionID string
+}
+
+// Name returns a name/version for this OS info, e.g. "centos 7.1"
+func (r OSRelease) Name() string {
+	return fmt.Sprintf("%v %v", r.ID, r.VersionID)
+}
+
+func getRealOSRelease() (info *OSRelease, err error) {
+	return getOSReleaseFromFiles(releases, versions)
+}
+
+func getOSReleaseFromFiles(releases, versions []string) (info *OSRelease, err error) {
+	release, err := openFirst(releases...)
+	if err != nil {
+		log.Warnf("Failed to read any release file: %v.", err)
+		return nil, trace.NotFound("no release version file found")
+	}
+
+	version, err := openFirst(versions...)
+	if err != nil {
+		log.Warnf("Failed to read any release version file: %v.", err)
+		// fallthrough
+	}
+
+	return getOSRelease(releaseFiles{
+		release:    release,
+		version:    version,
+		lsbRelease: lsbRelease,
+	})
+}
+
+func getOSRelease(files releaseFiles) (info *OSRelease, err error) {
+	// Try lsb_release first
+	if files.lsbRelease != nil {
+		info, err = files.lsbRelease()
+		if err == nil {
+			return info, nil
+		}
+	}
+
+	info, err = files.releaseInfo()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to determine release version")
+	}
+
+	if info == nil {
+		// At this point, we cannot determine release information
+		return nil, trace.NotFound("no release information found")
+	}
+
+	files.updateInfo(info)
+
+	return info, nil
+}
+
+// versionsMatch tests if test is equivalent to info
+func versionsMatch(test, info OSRelease) bool {
+	if strings.ToLower(test.ID) != strings.ToLower(info.ID) {
+		return false
+	}
+
+	// Versions are matched as prefixes, e.g. if a required version is 7.2 then
+	// it matches 7.2, 7.2.1, 7.2.2, etc.
+	if !strings.HasPrefix(info.VersionID, test.VersionID) {
```

Comment:
I have a tiny concern that this could false positive (7.1 would match 7.10), but I can't think of any mainstream linux distro which would trigger the false positive. Might be worth keeping in mind though.